packages/core-data/src/hooks/use-permissions.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { IDLE, SUCCESS, RESOLVING } from './constants';
+import { store as coreStore } from '../';
+
+export default function usePermissions( resource, id ) {
+	return useQuerySelect(
+		( resolve ) => {
+			const { canUser } = resolve( coreStore );
+			const create = canUser( 'create', resource );
+			let update, _delete;
+			if ( id ) {
+				update = canUser( 'update', resource, id );
+				_delete = canUser( 'delete', resource, id );
+			}
+			const isResolving =
+				create.isResolving ||
+				update?.isResolving ||
+				_delete?.isResolving;
+
+			const hasResolved =
+				create.hasResolved &&
+				( id ? update.hasResolved : true ) &&
+				( id ? _delete.hasResolved : true );
+
+			let status = '';
+			if ( isResolving ) {
+				status = RESOLVING;
+			} else if ( hasResolved ) {
+				status = SUCCESS;
+			} else {
+				status = IDLE;
+			}
+			return {
+				status,
+				isResolving,
+				hasResolved,
+				canCreate: create.data,
+				canUpdate: update?.data,
+				canDelete: _delete?.data,
```

Comment:
Based on the usage in https://github.com/WordPress/gutenberg/pull/38289, it might be convenient to assume that permissions return value value only when resolved:

```suggestion
				canCreate: hasResolved  && create.data,
				canUpdate: hasResolved && update?.data,
				canDelete: hasResolved && _delete?.data,
```

This way you still have granular control over the status but you don't need to guard values.

Reply:
I think this makes sense. We should assume that users don't have permission until a query is resolved.